### ISSUE TITLE
Different clusters depending on order of GeoJSON features

### ISSUE THREAD

**Issue by palhal** [mapbox/mapbox-gl-js#12142]
If this is expected, how should i sort my data before clustering it to get a predictable result? By the way, thanks for a great library!

**mapbox-gl-js version**: 2.9.2

**browser**: Chrome 104.0.5112.81

### Steps to Trigger Behavior
1. Add a GeoJSON source with clustering enabled
2. Add a layer for presenting the resulting clusters
3. Observe the current clusters on the map
4. Modify the source so that the data in the _features_ array are reordered
5. Observe that the clustering is applied differently than in step 3

### Link to Demonstration
https://jsfiddle.net/pkhalle/km3r6tc7/

 1. Fill in your access token and click Run
 2. Observe that there is a cluster with two points and a single unclustered point
 3. Uncomment the commented line in the _onLoad_ function
 4. Run the code again
 5. Observe that there is a cluster with three points and no unclustered points

### Expected Behavior
Clusters are consistent regardless of order in JSON array

### Actual Behavior
Clusters are inconsistent depending on order in JSON array

**Left:** Original order
**Right**: Reversed order
![cluster-2](https://user-images.githubusercontent.com/56862114/183447245-a7711526-d417-4675-9c93-6c2063c2a3d0.png) ![cluster-3](https://user-images.githubusercontent.com/56862114/183447244-586d4d19-e1b4-4b53-bfbb-334d59633b99.png)

**Comment by mourner** [comment]
Clustering uses a greedy algorithm (pick a point, form a cluster with surrounding points, go to next unvisited point etc.), so it is order-dependent by design. 

> how should i sort my data before clustering it to get a predictable result?

You can use any order — the same order will produce the same set of clusters.

**Comment by palhal** [comment]
Thank you for the clarification. The current behavior makes sense with your explanation. I tried to find some information on the Mapbox website but couldn't find any. I did however find [your blog post](https://blog.mapbox.com/clustering-millions-of-points-on-a-map-with-supercluster-272046ec5c97) just now which was interesting.

My use case is having two different sources (same geometries but different properties) and I saw some strange behavior when they were not sorted equally. The simplest solution would be to provide all properties in one source, but the properties are fetched from different sites and I don't want either to be a bottleneck for the other. Sorting them equally should be feasible though.
